File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/cert/SwlCertManager.java
package i2f.swl.cert;

import i2f.swl.cert.data.SwlCert;
import i2f.swl.cert.data.SwlCertPair;

/**
 * @author Ice2Faith
 * @date 2024/8/8 15:37
 * @desc
 */
public interface SwlCertManager {
    SwlCert load(String certId, boolean server);

    void store(SwlCert cert, boolean server);

    default SwlCert loadServer(String certId) {
        return load(certId, true);
    }

    default SwlCert loadClient(String certId) {
        return load(certId, false);
    }

    default void storeServer(SwlCert cert) {
        store(cert, true);
    }

    default void storeClient(SwlCert cert) {
        store(cert, false);
    }

    default SwlCertPair loadPair(String certId) {
        SwlCert server = load(certId, true);
        SwlCert client = load(certId, false);
        return new SwlCertPair(server, client);
    }

    default void storePair(SwlCertPair pair) {
        store(pair.getServer(), true);
        store(pair.getClient(), false);
    }
}

File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/core/cert/SwlCertManager.js
import SwlCertPair from "./data/SwlCertPair";

/**
 * @return {SwlCertManager}
 * @constructor
 */
function SwlCertManager() {

}

/**
 * @param certId {String}
 * @param server {boolean}
 * @return {SwlCert}
 */
SwlCertManager.prototype.load = function (certId,server) {

}
/**
 * @param cert {SwlCert}
 * @param server {boolean}
 * @return {void}
 */
SwlCertManager.prototype.store = function (cert,server) {

}
/**
 * @param certId {String}
 * @return {SwlCert}
 */
SwlCertManager.prototype.loadServer = function (certId) {
    return this.load(certId,true)
}
/**
 * @param certId {String}
 * @return {SwlCert}
 */
SwlCertManager.prototype.loadClient = function (certId) {
    return this.load(certId,false)
}
/**
 * @param cert {SwlCert}
 * @return {void}
 */
SwlCertManager.prototype.storeServer = function (cert) {
    this.store(cert,true)
}
/**
 * @param cert {SwlCert}
 * @return {void}
 */
SwlCertManager.prototype.storeClient = function (cert) {
    this.store(cert,false)
}
/**
 *
 * @param certId {String}
 * @return {SwlCertPair}
 */
SwlCertManager.prototype.loadPair = function (certId) {
    let server = this.load(certId, true);
    let client = this.load(certId, false);
    return new SwlCertPair(server, client);
}
/**
 *
 * @param pair {SwlCertPair}
 * @return {void}
 */
SwlCertManager.prototype.storePair = function (pair) {
    this.store(pair.server, true);
    this.store(pair.client, false);
}

export default SwlCertManager
